app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/registration/RegisterWithEoriAndIdController.scala

```diff
@@ -162,17 +162,22 @@ class RegisterWithEoriAndIdController @Inject() (
   def pending(service: Service): Action[AnyContent] =
     authAction.ggAuthorisedUserWithEnrolmentsAction { implicit request => _: LoggedInUserWithEnrolments =>
       for {
-        subscriptionDetails <- cache.subscriptionDetails
-        _                   <- cache.saveSubscriptionDetails(subscriptionDetails)
-        processedDate <- cache.registerWithEoriAndIdResponse.map(
-          resp => languageUtils.Dates.formatDate(resp.responseCommon.processingDate.toLocalDate)
-        )
-        _ <- cache.remove
+        submissionCompleteData <- cache.submissionCompleteDetails
+        _                      <- cache.remove
+        _                      <- cache.saveSubmissionCompleteDetails(submissionCompleteData)
       } yield Ok(
         subscriptionOutcomePendingView(
-          subscriptionDetails.eoriNumber.getOrElse(throw DataUnavailableException("No EORI found in cache")),
-          processedDate,
-          subscriptionDetails.name,
+          submissionCompleteData.subscriptionDetails.getOrElse(
+            throw DataUnavailableException("No subscriptionDetails found within submissionCompleteData in cache")
+          ).eoriNumber.getOrElse(throw DataUnavailableException("No EORI found in cache")),
+          languageUtils.Dates.formatDate(
+            submissionCompleteData.processingDate.getOrElse(
+              throw DataUnavailableException("No processingDate found within submissionCompleteData in cache")
+            ).toLocalDate
+          ),
+          submissionCompleteData.subscriptionDetails.getOrElse(
+            throw DataUnavailableException("No subscriptionDetails found within submissionCompleteData in cache")
+          ).name,
```

Comment:
Sorry for the delay - could we refactor the above to avoid the use of exceptions?

```  
  def pending(service: Service): Action[AnyContent] =
    authAction.ggAuthorisedUserWithEnrolmentsAction { implicit request => _: LoggedInUserWithEnrolments =>
      for {
        submissionCompleteData <- cache.submissionCompleteDetails
        _                      <- cache.remove
        _                      <- cache.saveSubmissionCompleteDetails(submissionCompleteData)
      } yield displaySubscriptionOutcomePendingView(submissionCompleteData)
    }


  def displaySubscriptionOutcomePendingView(submissionCompleteData: SubmissionCompleteData) = {
  	val result = for {
  		details <- submissionCompleteData.subscriptionDetails
  		processingDate <- submissionCompleteData.processingDate
	} yield Ok(subscriptionOutcomePendingView(
          details.eoriNumber,
          languageUtils.Dates.formatDate(processingDate.toLocalDate),
          details.name,
          service
        ).withSession(newUserSession))

  	result.getOrElse{
          logger.warn("Subscription Complete Data not found for this session")
  	  Redirect(startPage)
  	}
  }

Reply:
My two reasons for the change are:
1) I can understand what's going on a bit better, the code got a bit harder to read having to do all of the getOrElse and exception throwing
2) We're handling the scenario rather than just sending the user to an error page and causing a pagerduty.

Now if someone tries to come to this page, but there's no "complete" date for them, we send them back to the start of the journey because we can safely assume they haven't actually completed it, or their session has timed out.